lib/wasix/src/os/task/process.rs

```diff
@@ -133,6 +141,22 @@ pub struct WasiProcessInner {
     /// Represents a checkpoint which blocks all the threads
     /// and then executes some maintenance action
     pub checkpoint: WasiProcessCheckpoint,
+    /// Referenced list of wakers that will be triggered
```

Comment:
Can be put all related fields on a nested struct to encupsulate things a bit better?
And move all related types and this struct to a submodule?
(including the `acquire_` methods)

Reply:
done